scripts/bond-calculator.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env python3
+import sys
+from bisect import bisect_left
+from datetime import datetime
+from decimal import Decimal
+from json import loads
+from optparse import OptionParser
+from statistics import quantiles, StatisticsError
+
+from dateutil.relativedelta import relativedelta
+from jmbase import EXIT_ARGERROR, jmprint, get_log, utxostr_to_utxo, EXIT_FAILURE
+from jmbitcoin import amount_to_sat, sat_to_btc
+from jmclient import FidelityBondMixin, add_base_options, load_program_config, jm_single, get_interest_rate
+
+DESCRIPTION = """Given either a Bitcoin UTXO in the form TXID:n
+(e.g., 0e3e2357e806b6cdb1f70b54c3a3a17b6714ee1f0e68bebb44a74b1efd512098:0)
+or an amount in either satoshi or bitcoin (e.g., 150000, 0.1, 10.123, 10btc),
+calculate fidelity bond values for all possible locktimes in a one-year period
+(12 months, you can change that with the `-m --months` option).
+By default it uses the values from your joinmarket.cfg,
+you can override these with the `-i --interest` and `-e --exponent` options.
+Additionally, you can export the orderbook from ob-watcher.py and use the data here
+with the `-o --orderbook` option, this will compare the results from this script
+with the fidelity bonds in the orderbook.
+"""
+
+log = get_log()
+
+
+def add_one_month(dt: datetime) -> datetime:
+    return dt + relativedelta(months=1)
+
+
+def main() -> None:
+    parser = OptionParser(
+        usage="usage: %prog [options] UTXO or amount",
+        description=DESCRIPTION,
+    )
+    add_base_options(parser)
+    parser.add_option(
+        "-i",
+        "--interest",
+        action="store",
+        type="float",
+        dest="interest",
+        help="Interest rate to use for fidelity bond calculation (instead of interest_rate config)",
+    )
+    parser.add_option(
+        "-e",
+        "--exponent",
+        action="store",
+        type="float",
+        dest="exponent",
+        help="Exponent to use for fidelity bond calculation (instead of bond_value_exponent config)",
+    )
+    parser.add_option(
+        "-m",
+        "--months",
+        action="store",
+        type="int",
+        dest="months",
+        help="For how many months to calculate the fidelity bond values, each month has its own stats (default 12)",
+        default=12,
+    )
+    parser.add_option(
```

Comment:
I'm guessing you already thought about this but we could do this without a file export ... but, it's another reason to fold it into ob-watcher which already does the quite obscure jobs needed to get the orderbook without a fully fledged JM bot ...

Reply:
Yeah, we discussed this a little earlier. The main problem I have with folding it into ob-watcher is requiring the ob-watcher machinery every time.
If we decide to go the way of a dedicated module, could we have both? That way this functionality would be everywhere, as dedicated script (using a `.json` file), as ob-watcher extra page, and as RPC (I'm not sure here what the best solution would be to pass orderbook data).